src/unclogged/core.clj

```diff
@@ -29,6 +29,26 @@
     "3164" MessageFormat/RFC_3164
     "5424" MessageFormat/RFC_5424))
 
+(defn ^:private ->syslog-msg
+  "Turns defaults + message map into a SyslogMessage."
+  [defaults message]
+  (let [{:keys [message
+                message-id
+                app-name
+                hostname
+                process-id
+                facility
+                severity]}
+        (merge defaults message)]
+    (doto (SyslogMessage.)
```

Comment:
err, it's actually `(proxy FancyClass [a b c] ...)` where [a b c] is the arguments to `FancyClass`. This creates an instance of an anonymous type that's a subclass of `FancyClass` to keep the type system happy.